tuf/ngclient/_internal/trusted_metadata_set.py

```diff
@@ -149,7 +149,7 @@ def update_root(self, data: bytes) -> Metadata[Root]:
         Returns:
             Deserialized and verified root ``Metadata`` object
         """
-        if self.timestamp is not None:
+        if Timestamp.type in self._trusted_set:
```

Comment:
Out of curiosity, have you considered
```suggestion
        if Timestamp.type in self:
```
? Given the custom `__getitem__`, we don't have to access the internal `_trusted_set` directly anywhere else in this class.

Reply:
For context, this mildly put me off at first, because I thought your patch exposes the internal `_trusted_set`. But I realized that it does not and that the instance behaves just like the internal variable.

Reply:
yeah, I went back and forth with that one -- maybe it would have been clearer with just `self`

Reply:
IMO it's fine to access the protected attribute inside the class. I was just confused, because updater instances also have a `_trusted_set`  attribute, but that one is actually a `TrustedMetadataSet` instance. :D